### ISSUE TITLE
Make Icon Accessible 

### ISSUE THREAD

**Issue by AlexGStapleton** [siteorigin/so-widgets-bundle#896]
The Icon Form field is currently not accessible. Screen readers will simply skip over the icon or potentially attempt to read the icon unicode which will result in a jumbled mess.

Related #806.

**Comment by AlexGStapleton** [comment]
We've introduced a title field that will help screenreaders know more about the icon.
